### PR TITLE
LC-743 sign-out functionality on IDM

### PR DESCRIPTION
**Pull request by jainmangleshco**
Adding a sign out functionality in IDM so signout
work independently than LPG-UI

### CODE REVIEW

File: src/main/resources/application.yml [hunk 1]
```diff
@@ -34,6 +34,8 @@ security:
 
 identity:
   url: ${OAUTH_URL:http://localhost:8080}
+  signoutUrl: ${OAUTH_URL:http://localhost:8080}/logout
```
Comment (neeraj-jain-cbo) [inline src/main/resources/application.yml:37]: Please rename the variable name OAUTH_URL to SIGNOUT_URL or LOGOUT_URL

File: src/main/resources/application.yml [hunk 2]
```diff
@@ -34,6 +34,8 @@ security:
...
+  returnToUrl: ${MGMT_LOGIN_URL:http://localhost:8081}/mgmt/login
```
Comment (neeraj-jain-cbo) [inline src/main/resources/application.yml:38]: Please rename the returnToUrl to mgmtLoginUrl

File: src/main/java/uk/gov/cshr/controller/UserController.java [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.beans.factory.annotation.Value;
+import uk.gov.cshr.service.security.IdentityService;
+
+import javax.servlet.http.HttpServletRequest;
+import java.security.Principal;
+
+@Slf4j
```
Comment (neeraj-jain-cbo) [inline src/main/java/uk/gov/cshr/controller/UserController.java:13]: There is no logging in the code so @Slf4 is not required. Please remove it.

File: src/main/java/uk/gov/cshr/controller/UserController.java [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+@Controller
+@PreAuthorize("hasPermission(returnObject, 'read')")
```
Comment (neeraj-jain-cbo) [inline src/main/java/uk/gov/cshr/controller/UserController.java:15]: I don't think this is required for logout? Please check and remove.

File: src/main/resources/application.yml (hunk 2 above)
Comment (neeraj-jain-cbo) [inline src/main/resources/application.yml:38]: Please rename MGMT_LOGIN_URL to IDM_LOGIN_URL

Comment (neeraj-jain-cbo) [inline src/main/resources/application.yml:38]: Please rename MGMT_LOGIN_URL to IDM_URL

File: src/main/java/uk/gov/cshr/controller/UserController.java (hunk 3 above)
Comment (jainmangleshco) [reply src/main/java/uk/gov/cshr/controller/UserController.java:13]: Code updated.

File: src/main/java/uk/gov/cshr/controller/UserController.java (hunk 4 above)
Comment (jainmangleshco) [reply src/main/java/uk/gov/cshr/controller/UserController.java:15]: Code updated.

File: src/main/resources/application.yml (hunk 2 above)
Comment (jainmangleshco) [reply src/main/resources/application.yml:38]: Code updated.

File: src/main/resources/application.yml [hunk 5]
```diff
@@ -34,6 +34,8 @@ security:
 
 identity:
   url: ${OAUTH_URL:http://localhost:8080}
+  signoutUrl: ${OAUTH_URL:http://localhost:8080}/logout
+  returnToUrl: ${IDM_URL:http://localhost:8081}/mgmt/login
```
Comment (neeraj-jain-cbo) [inline src/main/resources/application.yml:38]: returnToUrl can be renamed to idmLoginUrl to make it login name in sync with its value.

**Comment by neeraj-jain-cbo** [comment]
This PR should be raised against the idt-develop branch rather then master branch.

**Pull request merged**